### PR TITLE
Ensure MacCatalyst default http handler is the same as iOS

### PR DESCRIPTION
**Pull request by steveisok**

### CODE REVIEW

**Comment by steveisok** [comment]
For MacCatalyst, the Xamarin.iOS type forwards into Xamarin.MacCatalyst, so I believe this will work correctly.

**Comment by rolfbjarne** [comment]
> For MacCatalyst, the Xamarin.iOS type forwards into Xamarin.MacCatalyst, so I believe this will work correctly.
> 
> /cc @rolfbjarne

The Xamarin.iOS assembly shipped for Xamarin.MacCatalyst is a compatibility assembly so that existing assemblies built for Xamarin.iOS work as well as possible. New code should target Xamarin.MacCatalyst, and not rely on this compatibility behavior.

File: src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.MacCatalyst.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            // Default is to use the iOS native handler
+            if (!IsNativeHandlerEnabled())
+            {
+                return new HttpClientHandler();
+            }
+
+            if (handlerMethod == null)
+            {
+                Type? runtimeOptions = Type.GetType("ObjCRuntime.RuntimeOptions, Xamarin.iOS");
```
Comment (filipnavara) [inline src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.MacCatalyst.cs:22]: The assembly name needs to be updated

Comment (steveisok) [reply src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.MacCatalyst.cs:22]: I'm losing my mind... I updated the wrong file ;-)

**Review by filipnavara — APPROVED** [review]
FWIW LGTM

File: src/libraries/System.Net.Http/src/System.Net.Http.csproj [hunk 2]
```diff
@@ -36,9 +36,10 @@
     <Compile Include="System\Net\Http\Headers\KnownHeaders.cs" />
     <Compile Include="System\Net\Http\HttpBaseStream.cs" />
     <Compile Include="System\Net\Http\HttpClient.cs" />
-    <Compile Condition="'$(TargetsAndroid)' != 'true' and '$(TargetsiOS)' != 'true' and '$(TargetsTVOS)' != 'true'" Include="System\Net\Http\HttpClient.CreateDefaultHandler.cs" />
+    <Compile Condition="'$(TargetsAndroid)' != 'true' and '$(TargetsiOS)' != 'true' and '$(TargetsMacCatalyst)' != 'true' and '$(TargetsTVOS)' != 'true'" Include="System\Net\Http\HttpClient.CreateDefaultHandler.cs" />
     <Compile Condition="'$(TargetsAndroid)' == 'true'" Include="System\Net\Http\HttpClient.CreateDefaultHandler.Android.cs" />
-    <Compile Condition="'$(TargetsiOS)' == 'true'" Include="System\Net\Http\HttpClient.CreateDefaultHandler.iOS.cs" />
+    <Compile Condition="'$(TargetsiOS)' == 'true' or '$(TargetsMacCatalyst)' == 'true'" Include="System\Net\Http\HttpClient.CreateDefaultHandler.iOS.cs" />
```
Comment (rolfbjarne) [inline src/libraries/System.Net.Http/src/System.Net.Http.csproj:41]: You'll have to update/revert this condition to not build the iOS code for Mac Catalyst, now that you have specific Mac Catalyst code.

**Review by rolfbjarne — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**